src/lib.rs

```diff
@@ -30,7 +32,11 @@ impl Encode for Method {
         } else {
             0
         };
-        buf.push_byte(has_cooldown_byte);
+        let constraints_cnt_byte: u8 = match &self.constraints {
```

Comment:
nitpick: We are currently using the term "length" when referring to bytes, while "count" is used when referring to # of encoded objects